### PR TITLE
Fix/zios 9075 setting  username text field has wrong characters when using chinese cangjie input method

### PR DESCRIPTION
**Pull request by billypchan**
tested for both Chinese and English input with space cases.

### CODE REVIEW

**Comment by typfel** [comment]
Could explain why these changes solves the bug? Just reading the code, it looks like the behaviour is identical to before this changes. Shouldn't `replacingOccurrences` do nothing if there's no matches so the only difference is that we aren't re-setting the `self.text` property...

**Comment by billypchan** [comment]
@typfel every time setting self.text property deselects the text field and chinese cangjie input method keyboard can not clean up the cangjie coding.

File: Wire-iOS/Sources/UserInterface/Settings/TailEditingTextField.swift [hunk 1]
```diff
@@ -46,11 +46,15 @@ class TailEditingTextField: UITextField {
         self.addTarget(self, action: #selector(TailEditingTextField.replaceNonBreakingSpacesWithNormalSpaces), for: UIControlEvents.editingDidEnd)
     }
     
-    func replaceNormalSpacesWithNonBreakingSpaces() {
-        self.text = self.text?.replacingOccurrences(of: type(of: self).normalSpace, with: type(of: self).nonBreakingSpace)
+    func replaceNormalSpacesWithNonBreakingSpaces(_ sender: Any) {
+        if let isContainsNormalSpace = (self.text?.contains(type(of: self).normalSpace)), isContainsNormalSpace {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Settings/TailEditingTextField.swift]: could be turned into a guard

File: Wire-iOS/Sources/UserInterface/Settings/TailEditingTextField.swift [hunk 2]
```diff
@@ -46,11 +46,15 @@ class TailEditingTextField: UITextField {
...
+            self.text = self.text?.replacingOccurrences(of: type(of: self).normalSpace, with: type(of: self).nonBreakingSpace)
+        }
     }
     
     func replaceNonBreakingSpacesWithNormalSpaces() {
-        self.text = self.text?.replacingOccurrences(of: type(of: self).nonBreakingSpace, with: type(of: self).normalSpace)
+        if let isContainsNonBreakingSpace = (self.text?.contains(type(of: self).nonBreakingSpace)), isContainsNonBreakingSpace {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Settings/TailEditingTextField.swift]: could be turned into a guard

**Pull request merged**